### PR TITLE
Add table aws_api_gateway_domain_name. closes #1581

### LINKED ISSUE #1581: Add table aws_api_gateway_domain_name
When I tried to query for a custom domain name for the rest API gateway, and got the following error.

```SQL
> select * from aws_api_gateway_domain_name
Error: relation "aws_api_gateway_domain_name" does not exist (SQLSTATE 42P01)
```

According to the terraform AWS provider, they call different paths in `aws-sdk-go` as follows.
- aws_apigateway_domain_name [here](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/apigateway/domain_name.go#L11)
- aws_api_gatewayv2_domain_name [here](https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/apigatewayv2/domain_name.go#L11)

I don't know if it's called the same API endpoint for both API gateways.
Could you please have a look?

**References**
- [Terraform resources](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_domain_name)
- [AWS documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/how-to-custom-domains.html)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
We need a certificate validation for creating a AWS Apigateway Domain Name which is a manual interruption, so ontegration test suit could not be added for this table.

Before the Amazon certificate authority (CA) can issue a certificate for your site, AWS Certificate Manager (ACM) must prove that you own or control all of the domain names that you specify in your request. You can choose to prove your ownership with either Domain Name System (DNS) validation or with email validation at the time you request a certificate. Validation applies only to publicly trusted certificates issued by ACM. ACM does not validate domain ownership for imported certificates or for certificates signed by a private CA.
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_api_gateway_domain_name
+-------------------------------+-----------------+------------------+---------------------------+--------------------------+-----------------------------+--------------------+--------------------------->
| domain_name                   | certificate_arn | certificate_name | certificate_upload_date   | distribution_domain_name | distribution_hosted_zone_id | domain_name_status | domain_name_status_message>
+-------------------------------+-----------------+------------------+---------------------------+--------------------------+-----------------------------+--------------------+--------------------------->
| gateway.helix.huerot.io       | <null>          | <null>           | 2022-11-02T11:04:16+05:30 | <null>                   | <null>                      | AVAILABLE          | <null>                    >
| gateway.shkcdrwater.huerot.io | <null>          | <null>           | 2020-04-01T06:18:54+05:30 | <null>                   | <null>                      | AVAILABLE          | <null>                    >
+-------------------------------+-----------------+------------------+---------------------------+--------------------------+-----------------------------+--------------------+--------------------------->



> SELECT
  d.domain_name,
  config ->> 'CertificateArn' AS certificate_arn,
  c.certificate,
  c.certificate_transparency_logging_preference,
  c.created_at,
  c.imported_at,
  c.issuer,
  c.issued_at
FROM
  aws_api_gatewayv2_domain_name AS d
  CROSS JOIN jsonb_array_elements(d.domain_name_configurations) AS config
  LEFT JOIN aws_acm_certificate AS c ON c.certificate_arn = config ->> 'CertificateArn';

+-------------------------------+------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------>
| domain_name                   | certificate_arn                                                                          | certificate                                                      | certificate>
+-------------------------------+------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------>
| gateway.lskid.mkueof.io       | arn:aws:acm:ap-southeast-1:112233445566:certificate/c894a005-b8fa-450e-b2d6-196221058300 | -----BEGIN CERTIFICATE-----                                      | ENABLED    >
|                               |                                                                                          | MIIFzjCCBLagAwIBAgIQAnHeRIlt/CzcqdrAMPopyTANBgkqhkiG9w0BAQsFADA8 |            >
|                               |                                                                                          | MQswCQYDVQQGEwJVUzEPMA0GA1UEChMGQW1hem9uMRwwGgYDVQQDExNBbWF6b24g |            >
|                               |                                                                                          | UlNBIDIwNDggTTAxMB4XDTIzMDIwMjAwMDAwMFoXDTIzMDgwOTIzNTk1OVowHDEa |            >
|                               |                                                                                          | MBgGA1UEAwwRKi5oZWxpeC50dXJib3QuaW8wggEiMA0GCSqGSIb3DQEBAQUAA4IB |            >
|                               |                                                                                          | DwAwggEKAoIBAQC3+6pSefDZF2Enp2BdRZjVj+gYTQ+VRKFsnpNmAkfKNrPK7K2B |            >
|                               |                                                                                          | s7g02hALiecynZUisfskn7ElV6Bpq9605HCCftpNThnTHFr2MHROBWWuEZxXug3z |            >
|                               |                                                                                          | xd4P27kyOEgG8fcCIpXFpmCyqmevXvnIGIhyczatUxD1SZ7IJDXKuMpdjhPEDo5h |            >
|                               |                                                                                          | 1mAImiDdGGTyxcsgWrqCnKx7J1rBcU1T3qk0bskQAxaXeDstajfH5fyjwETopCr2 |            >
|                               |                                                                                          | XP4b1ikLmIVxpzkVMbNRl14+EuhXKx9S0Zuzpn4SGNela/5P9CMcr5xdUSy+5lNo |            >
|                               |                                                                                          | H+sYtMvzay9+D7qsY6ujROIW3fnXgVimjjGtAgMBAAGjggLqMIIC5jAfBgNVHSME |            >
|                               |                                                                                          | GDAWgBSBuA5jiokSGOX6OztQlZ/m5ZAThTAdBgNVHQ4EFgQU8feIlSzDJO3giEh0 |            >
|                               |                                                                                          | 61gjauAm6a0wHAYDVR0RBBUwE4IRKi5oZWxpeC50dXJib3QuaW8wDgYDVR0PAQH/ |            >
|                               |                                                                                          | BAQDAgWgMB0GA1UdJQQWMBQGCCsGAQUFBwMBBggrBgEFBQcDAjA7BgNVHR8ENDAy |            >
|                               |                                                                                          | MDCgLqAshipodHRwOi8vY3JsLnIybTAxLmFtYXpvbnRydXN0LmNvbS9yMm0wMS5j |            >
|                               |                                                                                          | cmwwEwYDVR0gBAwwCjAIBgZngQwBAgEwdQYIKwYBBQUHAQEEaTBnMC0GCCsGAQUF |            >
|                               |                                                                                          | BzABhiFodHRwOi8vb2NzcC5yMm0wMS5hbWF6b250cnVzdC5jb20wNgYIKwYBBQUH |            >
|                               |                                                                                          | MAKGKmh0dHA6Ly9jcnQucjJtMDEuYW1hem9udHJ1c3QuY29tL3IybTAxLmNlcjAM |            >
|                               |                                                                                          | BgNVHRMBAf8EAjAAMIIBfgYKKwYBBAHWeQIEAgSCAW4EggFqAWgAdwDoPtDaPvUG |            >

```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
The PR is still in draft, because I am unable to test the table due to unable to create the resource in console. We need to have proper resource setups for it.

- We need to have a valid certificate for it(We will be charge for a month at least).
- Certificate validation must have done.
- Route53 zone with appropriate record set.

File: aws/table_aws_api_gateway_domain_name.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+			d.StreamListItem(ctx, domain)
+
+			// Context may get cancelled due to manual cancellation or if the limit has been reached
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, err
```
Comment (karanpopat) [inline aws/table_aws_api_gateway_domain_name.go:193]:
```suggestion
	return nil, nil
```

File: aws/table_aws_api_gateway_domain_name.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+	switch h.Item.(type) {
+	case *types.DomainName:
+		domainName = *h.Item.(*types.DomainName).DomainName
+	case *apigateway.GetDomainNameOutput:
+		domainName = *h.Item.(*apigateway.GetDomainNameOutput).DomainName
+	}
+
+	commonData, err := getCommonColumns(ctx, d, h)
+	if err != nil {
+		return nil, err
```
Comment (karanpopat) [inline aws/table_aws_api_gateway_domain_name.go:248]: Should we add a logger here?

File: docs/tables/aws_api_gateway_domain_name.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  certificate_upload_date,
+  regional_certificate_arn,
+  domain_name_status
+from
+  aws_api_gateway_domain_name
+where
+  domain_name_status = 'AVAILABLE';
+```
+
+### Get certificate details of each domain names
```
Comment (karanpopat) [inline docs/tables/aws_api_gateway_domain_name.md:36]:
```suggestion
### Get certificate details of each domain name
```

**Review by karanpopat — COMMENTED** [review]
@ParthaI please check review comments

**Review by misraved — APPROVED** [review]

**Pull request merged**